azurelinuxagent/daemon/env.py

```diff
@@ -100,6 +107,33 @@ def handle_dhclient_restart(self):
            self.dhcp_handler.conf_routes()
            self.dhcpid = newpid
 
+    def handle_fstab_update(self):
+        """
+        Look for changes in fstab, mount if the fstab has new modified stamp 
+        If error, only re-check every minute to give an admin
+         time to correct it and avoid filling the log
+        """
+        fstabCurrentModifiedStamp = os.path.getmtime("/etc/fstab")
+        if fstabCurrentModifiedStamp != self.fstabModTime and \
+                            datetime.datetime.now() >  \
+                            self.lastNotice + datetime.timedelta(seconds=60):
+            ret, output = shellutil.run_get_output("mount -av")
```

Comment:
@hglkrijger - I agree with your sentiment about minimizing annoyance for users, but I do not see the same results in using -avf. I tried with the following versions provisioned from Azure VM marketplace images:
- RHEL 7.2
- CentOS 7
- Ubuntu 14.4
- Ubuntu 15.10
- Ubuntu 16.04

Using [fstabTests](https://github.com/Azure/WALinuxAgent/files/314265/fstabTests.txt), with the exception of Ubuntu 14.04 (which returned a success code for all tests) the invalid GUID was caught with -avf, but not the _invalid device name_ or _malformed entry_ test.

I am happy to test with your distro+version, but given these results I fear the change would have limited benefit using -avf in lieu of -av.  For example from CentOS 7:
```
# ./fstabTest.sh -avf
=============================================
Incorrect UUID
/                        : ignored
mount: can't find UUID=1111111-1111-1111-1111-111111111111
Error code: 32
=============================================
Invalid device name - no sdf1 on this computer
/                        : ignored
/testpartion             : successfully mounted
Error code: 0
=============================================
malformed entry - same as above but reversed device and partition name
/                        : ignored
/dev/sdf1                : successfully mounted
Error code: 0
=============================================
Additional entry - just to validate
/                        : ignored
/testpartion             : successfully mounted
Error code: 0

# ./fstabTest.sh -av
=============================================
Incorrect UUID
/                        : ignored
mount: can't find UUID=1111111-1111-1111-1111-111111111111
Error code: 32
=============================================
Invalid device name - no sdf1 on this computer
/                        : ignored
mount: special device /dev/sdf1 does not exist
Error code: 32
=============================================
malformed entry - same as above but reversed device and partition name
/                        : ignored
mount: mount point /dev/sdf1 does not exist
Error code: 32
=============================================
Additional entry - just to validate
/                        : ignored
/testpartion             : successfully mounted
Error code: 0

```